File: backend/spring-nfc/src/main/java/com/example/springnfc/entity/Etape.java
package com.example.springnfc.entity;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@ToString
public class Etape implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Lob
    private String comment;
    private String data;
    private String data2;
    private String validation;
    private String internal_validation;

    @ManyToOne
    private Intervention intervention;
}

File: frontend/flutter-nfc/lib/models/etape.dart
class Etape {
  int? id;
  String? comment;
  String? data;
  String? data2;
  String? internal_validation;
  String? validation;
  bool? completed;


  Etape({required this.id, required this.comment, required this.data,required this.data2, required this.internal_validation, required this.validation, required this.completed});

  Etape.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    comment = json['comment'];
    data = json['data'];
    data2 = json['data2'];
    internal_validation = json['internal_validation'];
    validation = json['validation'];
    completed = json['completed'];
  }


  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id.toString();
    data['comment'] = this.comment.toString();
    data['data'] = this.data.toString();
    data['data2'] = this.data2.toString();
    data['internal_validation'] = this.internal_validation.toString();
    data['validation'] = this.validation.toString();
    data['completed'] = this.completed.toString();
    return data;
  }
}